src/interface/sparse_pack.hpp

```diff
@@ -318,10 +398,27 @@ class SparsePack : public SparsePackBase {
   template <class TIn, REQUIRES(IncludesType<TIn, Ts...>::value)>
   KOKKOS_INLINE_FUNCTION Real &flux(const int b, const int dir, const TIn &t, const int k,
                                     const int j, const int i) const {
+    PARTHENON_DEBUG_REQUIRE(!flat_, "Accessor cannot be used for flat packs");
     PARTHENON_DEBUG_REQUIRE(dir > 0 && dir < 4 && with_fluxes_, "Bad input to flux call");
     const int vidx = GetLowerBound(b, t) + t.idx;
     return pack_(dir, b, vidx)(k, j, i);
   }
+
+ private:
+  // Must make a copy of the vector, since input vector is const,
+  // and it may not even be an lvalue.
+  static std::vector<MetadataFlag> AddFlag_(const std::vector<MetadataFlag> &flags,
+                                            MetadataFlag mf = Metadata::WithFluxes) {
```

Comment:
This change was suggested by @lroberts36 offline. It lets us omit the `Metadata::WithFluxes` flag when calling `GetWithFluxes`, as it should be implied.